internal/querycoord/segment_allocator.go

```diff
@@ -80,21 +77,23 @@ func shuffleSegmentsToQueryNode(ctx context.Context, reqs []*querypb.LoadSegment
 
 			for _, req := range reqs {
 				sort.Slice(nodeIDSlice, func(i, j int) bool {
-					return nodeID2NumSegemnt[nodeIDSlice[i]] < nodeID2NumSegemnt[nodeIDSlice[j]]
+					return nodeID2NumSegment[nodeIDSlice[i]] < nodeID2NumSegment[nodeIDSlice[j]]
 				})
 				req.DstNodeID = nodeIDSlice[0]
-				nodeID2NumSegemnt[nodeIDSlice[0]]++
+				nodeID2NumSegment[nodeIDSlice[0]]++
 			}
 			return nil
 		}
 
 		if !wait {
-			return errors.New("no queryNode to allocate")
+			err := errors.New("no queryNode to allocate")
```

Comment:
this error information lack of information.
why no querynode?
1) dead?
2) lack of memory?
3) or ..

Reply:
it will help on locating the root cause

Reply:
more error info has been added